File: my-folder/0020-valid-parentheses/solution.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        openners = []
        objects = {"}":"{", ")":"(", "]":"["}
        for i in s:
            if i in objects:
                if openners and openners[-1] == objects[i]:
                    openners.pop()
                else:
                    return False
            else:
                openners.append(i)
        return True if not openners else False 

File: my-folder/0020-valid-parentheses/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = []
    const map = new Map([
        ['(', ')'],
        ['{', '}'],
        ['[', ']'],

    ])

    for(let i =0; i<s.length; i++){
        if(s[i] == '(' || s[i] =='{' || s[i]=='['){
            stack.push(s[i]);
        }else{
            const last = stack.pop()
            if (s[i] !== map.get(last)){
                return false
            }
        }
    }
    return stack.length === 0;
};
